server/src/main/java/org/spine3/server/entity/FieldMasks.java

```diff
@@ -0,0 +1,191 @@
+     * @see #isValid(FieldMask)
+     */
+    @SuppressWarnings({"MethodWithMultipleLoops", "unchecked"})
+    public static <M extends  Message, B extends Message.Builder> Collection<M> applyMask(@SuppressWarnings("TypeMayBeWeakened") FieldMask mask, Collection<M> entities, TypeUrl typeUrl) {
+        final List<M> filtered = new ArrayList<>();
+        final ProtocolStringList filter = mask.getPathsList();
+
+        final Class<B> builderClass = getBuilderForType(typeUrl);
+
+        if (filter.isEmpty() || builderClass == null) {
+            return Collections.unmodifiableCollection(entities);
+        }
+
+        try {
+            final Constructor<B> builderConstructor = builderClass.getDeclaredConstructor();
+            builderConstructor.setAccessible(true);
+
+            for (Message wholeMessage : entities) {
+                final B builder = builderConstructor.newInstance();
+
+                for (Descriptors.FieldDescriptor field : wholeMessage.getDescriptorForType().getFields()) {
+                    if (filter.contains(field.getFullName())) {
+                        builder.setField(field, wholeMessage.getField(field));
+                    }
+                }
+
+                filtered.add((M) builder.build());
+            }
+
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException e) {
+            // If any reflection failure happens, return all the data without any mask applied.
+            return Collections.unmodifiableCollection(entities);
+        }
+
+        return Collections.unmodifiableList(filtered);
+    }
+
+    /**
+     * <p>Checks whether the given {@code FieldMask} is valid of not.
+     * This also includes a null check.
+     *
+     * @param fieldMask Nullable {@code FieldMask} to check.
+     * @return {@code true} if the {@code FieldMask} is valid for use, {@code} false otherwise.
+     */
+    public static boolean isValid(FieldMask fieldMask) {
+        //noinspection ConstantConditions - redundant null check.
+        return fieldMask != null && !fieldMask.getPathsList().isEmpty();
+    }
+
+    /**
+     * <p>Applies given {@code FieldMask} to a single {@link Message}.
+     *
+     * <p>The {@code FieldMask} must be valid for this operation.
+     *
+     * <p>In case of failure (e.g. wrong {@code TypeUrl} in the field mask) this method silently handles it
+     * without any exception or error thrown.
+     *
+     * @param mask {@code FieldMask} instance to apply.
+     * @param entity The {@link Message} to apply given {@code FieldMask} to.
+     * @param typeUrl Type of the {@link Message}.
+     * @return A {@link Message} of the same type as the given one with only selected fields.
+     * @see #isValid(FieldMask)
+     */
+    @SuppressWarnings("unchecked")
+    public static <M extends  Message, B extends Message.Builder> M applyMask(@SuppressWarnings("TypeMayBeWeakened") FieldMask mask, M entity, TypeUrl typeUrl) {
```

Comment:
The argument order should be `entity` (what to use), `fieldMask` (what to apply) and `typeUrl` (what to use for application).

Let's apply here and in related cases.